python/ray/data/tests/test_dataset_image.py

```diff
@@ -1,193 +1,142 @@
+
+        ds = ray.data.read_images(root, partitioning=partitioning)
+
+        assert ds.schema().names == ["image", "label"]
+
+        image_type, label_type = ds.schema().types
+        assert isinstance(image_type, ArrowTensorType)
+        assert pa.types.is_string(label_type)
+
+        df = ds.to_pandas()
+        assert sorted(df["label"]) == ["cat", "cat", "dog"]
+        assert all(tensor.numpy_shape == (32, 32, 3) for tensor in df["image"])
+
+    def test_e2e_prediction(self, ray_start_regular_shared):
+        from ray.train.torch import TorchCheckpoint, TorchPredictor
+        from ray.train.batch_predictor import BatchPredictor
+
+        from torchvision import transforms
+        from torchvision.models import resnet18
+
+        dataset = ray.data.read_images("example://image-datasets/simple")
+
+        transform = transforms.ToTensor()
+        dataset = dataset.map(transform)
+
+        model = resnet18(pretrained=True)
+        checkpoint = TorchCheckpoint.from_model(model=model)
+        predictor = BatchPredictor.from_checkpoint(checkpoint, TorchPredictor)
+        predictor.predict(dataset)
+
+    @pytest.mark.parametrize(
+        "image_size,image_mode,expected_size,expected_ratio",
+        [(64, "RGB", 30000, 4), (32, "L", 3500, 0.5), (256, "RGBA", 750000, 85)],
     )
-    assert (
-        reader._encoding_ratio >= expected_ratio
-        and reader._encoding_ratio <= expected_ratio * 1.5
-    ), "encoding ratio is out of expected bound"
-    data_size = reader.estimate_inmemory_data_size()
-    assert (
-        data_size >= expected_size and data_size <= expected_size * 1.5
-    ), "estimated data size is out of expected bound"
+    def test_data_size_estimate(
+        self,
+        ray_start_regular_shared,
+        image_size,
+        image_mode,
+        expected_size,
+        expected_ratio,
+    ):
+        root = "example://image-datasets/different-sizes"
+        ds = ray.data.read_images(
+            root, size=(image_size, image_size), mode=image_mode, parallelism=1
+        )
+
+        data_size = ds.size_bytes()
+        assert data_size >= 0, "estimated data size is out of expected bound"
+        data_size = ds.fully_executed().size_bytes()
+        assert data_size >= 0, "actual data size is out of expected bound"
+
+        reader = _ImageDatasourceReader(
+            delegate=ImageDatasource(),
+            paths=[root],
+            filesystem=LocalFileSystem(),
+            partition_filter=ImageDatasource.file_extension_filter(),
+            partitioning=None,
+            size=(image_size, image_size),
+            mode=image_mode,
+        )
+        assert (
+            reader._encoding_ratio >= expected_ratio
+            and reader._encoding_ratio <= expected_ratio * 1.5
+        ), "encoding ratio is out of expected bound"
+        data_size = reader.estimate_inmemory_data_size()
+        assert data_size >= 0, "estimated data size is out of expected bound"
```

Comment:
looks like `paths` can be a list as well but i didn't see a test for it -- what's the expected behavior ?

Reply:
Should be easy to add a version of `test_basic` that explicitly lists the 3 simple image file names.